src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,313 @@
+        feature_strides (`List[int]`, *optional*, defaults to [4, 8, 16, 32]):
+            Feature strides corresponding to features generated from backbone network.
+        output_auxiliary_logits (`bool`, *optional*):
+            Should the model output its `auxiliary_logits` or not.
+
+    Examples:
+
+    ```python
+    >>> from transformers import Mask2FormerConfig, Mask2FormerModel
+
+    >>> # Initializing a Mask2Former facebook/mask2former-instance-swin-small-coco configuration
+    >>> configuration = Mask2FormerConfig()
+
+    >>> # Initializing a model (with random weights) from the facebook/mask2former-instance-swin-small-coco style configuration
+    >>> model = Mask2FormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "mask2former"
+    attribute_map = {"hidden_size": "mask_feature_size"}
+
+    def __init__(
+        self,
+        backbone_config: Optional[Dict] = None,
+        decoder_config: Optional[Dict] = None,
+        ignore_value: Optional[int] = 255,
+        num_queries: Optional[int] = 150,
+        no_object_weight: Optional[float] = 0.1,
+        class_weight: Optional[float] = 2.0,
+        mask_weight: Optional[float] = 5.0,
+        dice_weight: Optional[float] = 5.0,
+        train_num_points: Optional[int] = 12544,
+        oversample_ratio: Optional[float] = 3.0,
+        importance_sample_ratio: Optional[float] = 0.75,
+        init_std: Optional[float] = 0.02,
+        init_xavier_std: Optional[float] = 1.0,
+        use_auxiliary_loss: Optional[bool] = True,
+        feature_strides: Optional[List[int]] = [4, 8, 16, 32],
+        output_auxiliary_logits: Optional[bool] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if backbone_config is None:
+            logger.info("`backbone_config` is `None`. Initializing the config with the default `Swin` backbone.")
+            backbone_config = SwinConfig(
+                image_size=224,
+                in_channels=3,
+                patch_size=4,
+                embed_dim=96,
+                depths=[2, 2, 6, 2],
+                num_heads=[3, 6, 12, 24],
+                window_size=7,
+                drop_path_rate=0.3,
+                out_features=["stage1", "stage2", "stage3", "stage4"],
+            )
+
+        decoder_config_dict = kwargs.pop("decoder_config_dict", None)
+
+        if decoder_config_dict is not None:
+            decoder_config = decoder_config_dict
+
+        if decoder_config is None:
+            decoder_config = {}
+            logger.info("decoder_config is None. Initializing the Mask2FormerDecoderConfig with default values.")
+
+        self.backbone_config = backbone_config
+
+        if isinstance(decoder_config, PretrainedConfig):
+            decoder_config = decoder_config.to_dict()
+
+        self.decoder_config = Mask2FormerDecoderConfig(**decoder_config)
```

Comment:
Could this be cleaned up a bit? I don't see why we accept both decoder_config_dict and decoder_config.

Reply:
Hi Niels, I referred [clipsegconfig](https://github.com/huggingface/transformers/blob/main/src/transformers/models/clipseg/configuration_clipseg.py#:~:text=text_config_dict%20%3D%20kwargs.pop(%22text_config_dict%22%2C%20None)) to make above change. I was a bit confused by this too but in a hurry thought of clarifying this later. @alaradirik has cleaned this up now. But any reason why it's used in clipseg though?